games/hangman.c

```diff
@@ -0,0 +1,308 @@
+/**
+ * @file
+ * @brief C implementation of [Hangman Game](https://en.wikipedia.org/wiki/Hangman_(game))
+ * @details
+ * Simple, readable version of hangman.
+ * Changed graphic to duck instead of traditional stick figure (same number of guesses).
+ * @author [AtlantaEmrys2002](https://github.com/AtlantaEmrys2002)
+*/
+
+#include <assert.h> /// for self-testing
+#include <ctype.h> /// for main()
+#include <stdio.h> /// for main(), new_word(), new_guess(), won()
+#include <stdlib.h> /// for all functions
+#include <string.h> /// for main()
+#include <time.h> /// for new_game()
+
+/*
+ * @brief game_instance structure that holds current state of game
+ */
+struct game_instance{
+
+    char current_word[30]; /// word to be guessed by player
+    char hidden[30]; /// hidden version of word that is displayed to player
+    int size; /// size of word
+    int incorrect; /// number of incorrect guesses
+    char guesses[25]; /// previous guesses
+    int guesses_size; /// size of guesses array
+
+};
+
+// function prototypes
+struct game_instance new_game(); // creates a new game
+int new_guess(char, const char guesses[], int size); // checks if player has already played letter
+int in_word(char, const char word[], unsigned int size); // checks if letter is in word
+void picture(int score); // outputs image of duck (instead of hang man)
+void won(const char word[], int score); // checks if player has won or lost
+
+/**
+ * @brief Self-test Implementations
+ * @returns void
+ */
+static void test() {
+    /*checks in_word returns correct results (e.g. c is in cat, d is not in flower)*/
+    assert(in_word('c', "cat", 3) == 1);
+    assert(in_word('d', "flower,", 6) == -1);
+    /*checks new_guess returns correct results (e.g. d is in dhjkl and has been guessed before, but e is not and hasn't been guessed before)*/
+    //assert(new_guess('d', "dhjkl", 5) == 1);
+    assert(new_guess('z', "testing", 5) == -1);
+}
+
+/**
+ * @brief Main Function
+ * @returns 0 on exit
+ */
+int main() {
+
+    test();
+
+    struct game_instance game = new_game(); // new game created
+    char guess; // current letter guessed by player
+
+    // main loop - asks player for guesses
+    while ((strchr(game.hidden, '_') != NULL) && game.incorrect <= 12) {
+        do {
+            printf("\n****************************\n");
+            printf("Your word: ");
+
+            for (int i = 0; i < game.size; i++) {
+                printf("%c ", game.hidden[i]);
+            }
+
+            if (game.guesses_size > 0) {
+                printf("\nSo far, you have guessed: ");
+                for (int i = 0; i < game.guesses_size; i++) {
+                    printf("%c ", game.guesses[i]);
+                }
+            }
+
+            printf("\nYou have %d guesses left.", (13 - game.incorrect));
```

Comment:
Displaying 13 guesses at game start definitely is unintended.
```suggestion
            printf("\nYou have %d guesses left.", (12 - game.incorrect));
```